### DISCUSSION TITLE
Completing functions snippets with tabstops

### DISCUSSION THREAD

**Discussion by tinny-blueleo** [hrsh7th/nvim-cmp#731] [Q&A] [answered]
Hey friends!  Hope you are well!  In the intro page for this repo in the video: Kapture.2021-10-27.at.13.32.44.mp4

You can see a function being expanded and tab being used to quickly cycle through the function arguments to populate them.  Does anyone know how to get this functionality working and the list of plugins to do so?  I've tried stringing together nvim-cmp with snippy and with luasnip and can only tab through code snippets and not a function that I've written being called or even a function that's part of the standard library in the language (e.g. table.insert in lua which works in the demo video).

**Comment by hrsh7th** [comment, marked as answer]
My sumneko lua settings are the following.

```lua
  require'lspconfig'.sumneko_lua.setup {
    on_init = on_init;
    capabilities = capabilities,
    cmd = {
      vim.fn.expand('~/Develop/Repo/lua-language-server/bin/macOS/lua-language-server'),
      "-E",
      vim.fn.expand('~/Develop/Repo/lua-language-server/main.lua'),
    };
    settings = {
      Lua = {
        runtime = {
          version = 'LuaJIT',
        },
        completion = {
          callSnippet = 'Replace',
        },
        diagnostics = {
          enable = true,
          globals = {
            "vim", "describe", "it", "before_each", "after_each"
          }
        },
        workspace = {
          library = {
            vim.fn.expand('$VIMRUNTIME/lua'),
            vim.fn.expand('$VIMRUNTIME/lua/vim/lsp'),
          },
        },
      },
    },
  }
```

**Comment by tinny-blueleo** [comment]
That was it, specifically this setting:

settings = {Lua = {completion = {callSnippet = 'Replace'}}}

Fixed it and I was able to automatically fix lua function signatures with tabstops like in the demo.  Thank you so much!  Do you know if there's anyway this can be generalized to other languages or is this heavily dependent on the LSP server?

**Reply by hrsh7th** [reply]
It's completely dependent to LS.

**Reply by tinny-blueleo** [reply]
Got it, thanks a bunch friend!
